src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -10,6 +11,15 @@ namespace Microsoft.AspNet.Server.Kestrel.Infrastructure
 {
     public struct MemoryPoolIterator2
     {
+        // TODO: should be configurable? Default .NET 4.5 lengths are, configurable upwards:
+        // maxUrlLength = 260 (windows path length?)
+        // maxQueryStringLength = 2048
+        // Needs cooperation of reverse proxy if used
+        // In wild max cookie size ~4093 bytes http://stackoverflow.com/questions/640938/what-is-the-maximum-size-of-a-web-browsers-cookies-key
+        // In wild max url length ~2000 bytes http://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers
+        // Security issues around unbounded limits
+        private const int _maxHeaderLength = 16384;
```

Comment:
+1 - Though we would use this externally with no problems, there are some internal...optimizations...that involve large headers for us. 16k would break on them.